frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -0,0 +1,846 @@
+            Ok(())
+        }
+
+        /// Force there to be a new era at the end of blocks indefinitely.
+        ///
+        /// The dispatch origin must be Root.
+        ///
+        /// # Warning
+        ///
+        /// The election process starts multiple blocks before the end of the era.
+        /// If this is called just before a new era is triggered, the election process may not
+        /// have enough blocks to get a result.
+        ///
+        /// # <weight>
+        /// - Weight: O(1)
+        /// - Write: ForceEra
+        /// # </weight>
+        #[pallet::weight(T::WeightInfo::force_new_era_always())]
+        pub fn force_new_era_always(origin: OriginFor<T>) -> DispatchResult {
+            ensure_root(origin)?;
+            ForceEra::<T>::put(Forcing::ForceAlways);
+            Ok(())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        /// Update the ledger for a staker. This will also update the stash lock.
+        /// This lock will lock the entire funds except paying for further transactions.
+        fn update_ledger(staker: &T::AccountId, ledger: &StakingLedger<BalanceOf<T>>) {
+            if ledger.active.is_zero() {
+                Ledger::<T>::remove(&staker);
+                T::Currency::remove_lock(STAKING_ID, &staker);
+            } else {
+                T::Currency::set_lock(STAKING_ID, &staker, ledger.total, WithdrawReasons::all());
+                Ledger::<T>::insert(staker, ledger);
+            }
+        }
+
+        /// Checks if there is a valid smart contract for the provided address
+        fn is_contract_valid(address: &SmartContract<T::AccountId>) -> bool {
+            match address {
+                SmartContract::Wasm(_account) => {
+                    //     <pallet_contracts::ContractInfoOf<T>>::get(&account).is_some()
+                    false
+                }
+                SmartContract::Evm(_account) => {
+                    // pallet_evm::Module::<T>::account_codes(&account).len() > 0 TODO remove comment after EVM mege
+                    true
+                }
+            }
+        }
+
+        /// Calculate rewards for all stakers for this era
+        fn stakers_era_reward(
+            staker_map: &mut BTreeMap<T::AccountId, BalanceOf<T>>,
+            points: &EraStakingPoints<T::AccountId, BalanceOf<T>>,
+            reward_for_contract: BalanceOf<T>,
+        ) {
+            let staker_part = Perbill::from_rational(reward_for_contract, (*points).total);
+
+            for (s, b) in &points.stakers {
+                let reward = staker_map.entry(s.clone()).or_insert(Default::default());
+                *reward += staker_part * *b;
+            }
+        }
+
+        /// Execute payout for stakers
+        fn payout_stakers(staker_map: &BTreeMap<T::AccountId, BalanceOf<T>>) {
+            for (s, b) in staker_map {
+                T::Currency::deposit_into_existing(&s, *b).ok();
+            }
+        }
```

Comment:
@Maar-io 
Perhaps this needs to be changed so that we transfer from dapps staking account to the staker?

Dapps rewards are deposited into dapps account so we need to payout those funds.